src/main/java/org/apache/bcel/classfile/ConstantPool.java

```diff
@@ -57,7 +57,7 @@ public ConstantPool(Constant[] constant_pool) {
      * @throws IOException
      * @throws ClassFormatException
      */
-    ConstantPool(DataInput input) throws IOException, ClassFormatException {
+    public ConstantPool(DataInput input) throws IOException, ClassFormatException {
```

Comment:
The reason ConstantPool() should be public is explained here: https://issues.apache.org/jira/browse/BCEL-180

<blockquote>
In some cases such as JDI API we have the class constant pool content only in form of separate byte array. Currently there is no way to load constant pool from byte array.
<br /><br />
The proposed patch enable loading ConstantPool from custom DataInput stream by making ConstantPool(DataInput) constructor public.
</blockquote>